crates/shared/src/lib.rs

```diff
@@ -1 +1,22 @@
+use crate::consts::EXPECTED_RPC_VERSION;
+use anyhow::{Context, Result};
+use semver::Version;
+use starknet::providers::{jsonrpc::HttpTransport, JsonRpcClient, Provider};
+
 pub mod consts;
+
+#[must_use]
+pub fn is_supported_version(version: &Version) -> bool {
+    let expected_version =
+        Version::parse(EXPECTED_RPC_VERSION).expect("Failed to parse the expected RPC version");
+    *version == expected_version
+}
+
+pub async fn get_and_parse_spec_version(client: &JsonRpcClient<HttpTransport>) -> Result<Version> {
```

Comment:
```suggestion
pub async fn get_rpc_version(client: &JsonRpcClient<HttpTransport>) -> Result<Version> {
```
[nit] The signature already tells it's parsed